tests/unit/test_checkpointing.py

```diff
@@ -0,0 +1,93 @@
+from unittest.mock import MagicMock
+from unittest import TestCase
+from fenics import FunctionSpace, UnitSquareMesh, Expression, interpolate
+from fenicsprecice.solverstate import SolverState
+
+
+class TestCheckpointing(TestCase):
+    def test_solverstate_basic(self):
+        """
+        Check if correct values are read from the checkpoint, while the state of the object that is copied is not changed
+        """
+        time = 0.5
+        iteration = 1
+        size = 5
+        mesh = UnitSquareMesh(size,size)
+        V = FunctionSpace(mesh, 'P', 2)
+        E = Expression("t", t=time, degree=2)
+        u = interpolate(E, V)
+
+        # "write checkpoint"
+        sstate = SolverState(u, time, iteration)
+        # "read checkpoint"
+        u_cp, t_cp, n_cp = sstate.get_state()
+
+        #check values
+        self.assertEqual(t_cp, time)
+        self.assertEqual(iteration, n_cp)
+        #function should be the same everywhere (-> check vector values of the function)
+        vec_u = u.vector()
+        vec_u_cp = u_cp.vector()
+        for i in range(size*size):
+            self.assertAlmostEqual(vec_u[i], vec_u_cp[i])
+
+    def test_solverstate_modification_vector(self):
+        """
+        Check if correct values are read from the checkpoint, if the dof vector of the dolfin functions are changed directly 
+        """
+        time = 0.5
+        iteration = 1
+        size = 5
+        mesh = UnitSquareMesh(size,size)
+        V = FunctionSpace(mesh, 'P', 2)
+        E = Expression("t", t=time, degree=2)
+        u = interpolate(E, V)
+
+        # "write checkpoint"
+        sstate = SolverState(u, time, iteration)
+
+        # modify state of u
+        u.vector()[:] = time + 2
```

Comment:
See above. I think the following would be much easier to read:

```suggestion
        dummy_value += 2
        u.vector()[:] = dummy_value
```

Reply:
I would suggest to leave it like that, since the assertion still needs the original `dummy_value`. Then, I would need to subtract the 2 in the assertion instead.

Reply:
Then why not call it `reference_value` and `dummy_value`? You could then put a `reference_value = dummy_value` before you start modifying `dummy_value`.

But if you want to leave it as it is: Also ok.

Reply:
I saw this comment just now. I added the changes you suggested.

Reply:
Looks good.